x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/NativeRoleMappingStore.java

```diff
@@ -85,6 +84,17 @@
  */
 public class NativeRoleMappingStore implements UserRoleMapper {
 
+    /**
+     * This setting is never registered by the security plugin - in order to disable the native role APIs
+     * another plugin must register it as a boolean setting and cause it to be set to `false`.
```

Comment:
I'm not necessarily saying you should change anything, but I've been pretty open about saying "this is an extension point for serverless", because that context isn't obvious if you're just looking at the Elasticsearch codebase and a lot of the extension points look pretty weird without it. What do you think? (feel free to answer in another medium, I'm mostly thinking out loud here)

Reply:
Here's how I think about it:

In this particular case, my preferred option would've been to have this be a proper setting (i.e. register it for the xpack/security plugin), and choose to **not** document it.
But then there's this perspective, to which I agree in principle, that we should document everything, i.e. not have any hidden behavior that we don't explain, and in fact not fully support. And then, we end up, pretty much where we are today, i.e. with very stuffy docs, with a lot of parameters, that make it hard to accomplish setups.

Given this general consensus, my default choice, in this particular case only, would've then been to make the `enabled` setting for the native role mapping system a proper **and documented** setting of the xpack/security plugin. In my mind it makes sense to have `enable` toggle for most things, though I acknowledge that dependencies can make it a hell to properly support it (but not really in this particular case).

I finally opted to follow the pattern that's being used in the role store, with the unregistered setting, that's then registered in the serverless/security plugin.
I finally compromised for consistency.